src/test/java/com/example/demo/service/PatientServiceTest.java

```diff
@@ -0,0 +1,351 @@
+                .email("test@gmail.com")
+                .password("pass")
+                .idCardNo("123")
+                .firstName(null)
+                .lastName("kowalski")
+                .phoneNumber("123456789")
+                .birthday(LocalDate.of(2000, 2, 17))
+                .build();
+        Patient currentPatient = buildPatient();
+        when(patientRepository.findByEmail(email)).thenReturn(Optional.of(currentPatient));
+        //when_then
+        PatientDataIsNullException resultException = assertThrows(PatientDataIsNullException.class, () -> patientService.editPatient(email, patientCommand));
+        //then
+        assertEquals("Patient fields cannot be null", resultException.getMessage());
+        assertEquals(HttpStatus.BAD_REQUEST, resultException.getHttpStatus());
+    }
+
+    @Test
+    void editPatient_newEmailIsReserved_exceptionThrown() {
+        //given
+        String email = "test@gmail.com";
+        PatientCommand patientCommand = PatientCommand.builder()
+                .email("new@gmail.com")
+                .password("pass")
+                .idCardNo("123")
+                .firstName("jan")
+                .lastName("kowalski")
+                .phoneNumber("123456789")
+                .birthday(LocalDate.of(2000, 2, 17))
+                .build();
+        Patient patientToEdit = buildPatient();
+        when(patientRepository.findByEmail(email)).thenReturn(Optional.of(patientToEdit));
+        when(patientRepository.existsByEmail(patientToEdit.getEmail())).thenReturn(true);
+        //when_then
+        PatientAlreadyExistException resultException = assertThrows(PatientAlreadyExistException.class, () -> patientService.editPatient(email, patientCommand));
+        //then
+        assertEquals(HttpStatus.BAD_REQUEST, resultException.getHttpStatus());
+        assertEquals("Patient with email: new@gmail.com already exist", resultException.getMessage());
+    }
+
+    @Test
+    void editPassword_patientExists_returnPatientDTO() {
+        //given
+        String email = "test@gmail.com";
+        PasswordRequest passwordRequest = new PasswordRequest("newPass");
+        Patient currentPatient = buildPatient();
+        Patient updatedPatient = Patient.builder()
+                .id(1L)
+                .email("test@gmail.com")
+                .password("newPass")
+                .idCardNo("123")
+                .firstName("jan")
+                .lastName("kowalski")
+                .phoneNumber("123456789")
+                .birthday(LocalDate.of(2000, 2, 17))
+                .build();
+        when(patientRepository.findByEmail(email)).thenReturn(Optional.of(currentPatient));
+        when(patientRepository.save(any())).thenReturn(updatedPatient);
+        //when
+        PatientDTO result = patientService.editPassword(email, passwordRequest);
+        //then
+        assertEquals(1L, result.getId());
+        assertEquals("test@gmail.com", result.getEmail());
```

Comment:
mozesz sprawdzic przy uzyciu mockito.verify czy do metody save został przekazany pacjent z okreslonym nowym haslem -> ze wzgledu na implementacje metody equals dla encji to musimy napoiisac aby to pokryc odpowiedi argument matcher albo uzyc argument captor